src/parser/Quads.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright 2025, University of Freiburg
+// Chair of Algorithms and Data Structures
+// Author: Julian Mundhahs <mundhahj@tf.uni-freiburg.de>
+
+#include "Quads.h"
+
+// ____________________________________________________________________________________
+// Transform the triples and sets the graph on all triples.
+vector<SparqlTripleSimpleWithGraph> transformTriplesTemplate(
+    ad_utility::sparql_types::Triples triples,
+    const SparqlTripleSimpleWithGraph::Graph& graph) {
+  auto convertTriple = [&graph](const std::array<GraphTerm, 3>& triple)
+      -> SparqlTripleSimpleWithGraph {
+    return {triple[0].toTripleComponent(), triple[1].toTripleComponent(),
+            triple[2].toTripleComponent(), graph};
+  };
+
+  return ad_utility::transform(triples, convertTriple);
+}
+
+// Re-wraps the value into a variant `T` which has additional values.
+template <typename T>
+T expandVariant(const Quads::IriOrVariable& graph) {
```

Comment:
Function should be `static` or in the anonymous namespace if only local to a .cpp file.